ign_ros2_control/include/ign_ros2_control/ign_system_interface.hpp

```diff
@@ -81,11 +81,13 @@ class IgnitionSystemInterface
   /// related with the entity in Gazebo
   /// param[in] hardware_info structure with data from URDF.
   /// param[in] _ecm Entity-component manager.
+  /// param[in] update_rate controller update rate
   virtual bool initSim(
     rclcpp::Node::SharedPtr & model_nh,
     std::map<std::string, ignition::gazebo::Entity> & joints,
     const hardware_interface::HardwareInfo & hardware_info,
-    ignition::gazebo::EntityComponentManager & _ecm) = 0;
+    ignition::gazebo::EntityComponentManager & _ecm,
+    int & update_rate) = 0;
```

Comment:
Does this need to be a reference?  I think it makes the API kind of awkward.

Reply:
This value is read after the class is already initialize and we don't have access to the pointer because it was std::move to `ros2_control`. I know it's ugly, any idea about how to improve it ?

Reply:
Ah, I didn't see that was the sequence.  I can't think of a reasonable way to improve it.